java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -94,10 +95,41 @@ public JObject(
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
         String nextPrefix = (config.getPrefixStrategy() == Config.Prefix.ALWAYS) ? classPrefix : "";
         String nextSuffix = (config.getSuffixStrategy() == Config.Suffix.ALWAYS) ? classSuffix : "";
+        String fieldKey = field.getKey();
+        // let's handle duplicates
+        final Map<String, Map<String, JSONSchemaProps>> duplicates = new HashMap<>();
+        // counting the current field duplicates
+        final List<Map.Entry<String, JSONSchemaProps>> fieldDuplicates = fields.entrySet().stream()
+            .filter(f -> sanitizeString(field.getKey()).equals(sanitizeString(f.getKey())))
+            .collect(Collectors.toList());
+        final Long duplicatesCount = fieldDuplicates.stream().count();
+        final Boolean duplicatesExist = duplicatesCount > 1;
+        if (duplicatesExist) {
+          // we want to throw an exception on some duplicates missing requirements. At the moment, the only one we
+          // enforce is for a (duplicatesCount - 1) number of duplicates to be marked as deprecated
+          final Long deprecatedDuplicates = fieldDuplicates.stream()
+              .filter(d -> d.getValue().getDescription().startsWith(DEPRECATED_FIELD_MARKER)).count();
+          if (deprecatedDuplicates != (duplicatesCount - 1)) {
+            throw new JavaGeneratorException(
+                String.format("The %s field has %d duplicates while only %d of them are deprecated", field.getKey(),
+                    duplicatesCount, deprecatedDuplicates));
+          }
+          final String fieldName = sanitizeString(field.getKey());
+          Map<String, JSONSchemaProps> existingDuplicates = duplicates.get(fieldName);
+          if (existingDuplicates == null) {
+            existingDuplicates = new HashMap<>();
+            duplicates.put(fieldName, existingDuplicates);
+          }
+          if (field.getValue().getDescription().startsWith(DEPRECATED_FIELD_MARKER)) {
+            fieldKey += String.format("-deprecated%d", existingDuplicates.keySet().size());
```

Comment:
as an addition, we can also annotate the field with 'Deprecated'

Reply:
Can you share a pointer to some spec draft? I only found http://json-schema.org/draft/2020-12/json-schema-validation.html#name-deprecated (see https://github.com/json-schema-org/json-schema-spec/pull/737/files) which makes me think we could rely on this rather than on the description?

Reply:
I've never seen the `deprecated` field used in the CRDs sadly.
Here I was asking to use JavaParser to add the `@Deprecated` annotation on the Java field.